Lib/typing.py

```diff
@@ -1393,7 +1393,11 @@ def __mro_entries__(self, bases):
                 return ()
             i = bases.index(self)
             for b in bases[i+1:]:
-                if isinstance(b, _BaseGenericAlias) and b is not self:
+                if b is self:
+                    continue
+                if isinstance(b, _BaseGenericAlias):
```

Comment:
Both of these diffs would make all tests pass:

```suggestion
                if isinstance(b, (_BaseGenericAlias, types.GenericAlias)):
```

```suggestion
                if isinstance(b, types.GenericAlias):
                    b = b.__origin__
                if isinstance(b, _BaseGenericAlias):
```

_But_, they can't both be right ;)

Reply:
The first breaks when you use `list` instead of `List` (see diff [here](https://github.com/python/cpython/pull/108415#discussion_r1307378946)) because it excludes `Generic` if any of the following base types is `types.GenericAlias` while we should only exclude it if it is both that and has `Generic` in its MRO.

The second might be correct, I can't say off the top of my head.

Not to say that I disagree on your conclusion that this indicates one or more test cases are missing.